src/gendersprachekorrigieren.ts

```diff
@@ -655,62 +189,67 @@ export class BeGone {
         return new Replacement(String.raw`(\n|\r|\r\n)`, "ig", " ", "").replace(s, counter);
     }
 
-    private applyToNodes(nodes: Array<CharacterData>, modifyData: (s: string) => string) {
-        var textnodes = nodes;
-        for (var i = 0; i < textnodes.length; i++) {
-            var node = textnodes[i];
-            var oldText = node.data;
-            var newText = oldText;
+    private applyToNodes(nodes: Array<CharacterData>, modifyData: (this: void, s: string) => string) {
+        const textnodes = nodes;
+        for (let i = 0; i < textnodes.length; i++) {
+            const node = textnodes[i];
+            const oldText = node.data;
+            let newText = oldText;
 
-            var parentNodeName = node.parentNode ? node.parentNode.nodeName.toLowerCase() : "";
+            const parentNodeName = node.parentNode ? node.parentNode.nodeName.toLowerCase() : "";
             // special treatment of HTML nodes that are only there for formatting; those might tear a word out of it's context which is important for correcting
             if (this.isHTMLFormattingNodeName(parentNodeName)) {
                 // this word needs to be replaced in context
-                var oldTextInContext = (i > 0 ? textnodes[i - 1].data : "") + "\f" + oldText + "\f" + (i < textnodes.length - 1 ? textnodes[i + 1].data : "");
+                let oldTextInContext = (i > 0 ? textnodes[i - 1].data : "") + "\f" + oldText + "\f" + (i < textnodes.length - 1 ? textnodes[i + 1].data : "");
                 //oldTextInContext = this.replaceLineBreak(oldTextInContext);
-                oldTextInContext = modifyData.call(this, oldTextInContext);
-                var index1 = oldTextInContext.indexOf("\f");
-                var index2 = oldTextInContext.indexOf("\f", index1 + 1);
-                var index3 = oldTextInContext.indexOf("\f", index2 + 1);
+                oldTextInContext = modifyData(oldTextInContext);
+                const index1 = oldTextInContext.indexOf("\f");
+                const index2 = oldTextInContext.indexOf("\f", index1 + 1);
+                const index3 = oldTextInContext.indexOf("\f", index2 + 1);
                 if (index1 > -1 && index2 > -2 && index3 === -1) // sanity check - RegEx magic might remove our marker; fall back to old behavior in this case
                 {
                     newText = oldTextInContext.substring(index1 + 1, index2);
                 } else {
                     //oldText = this.replaceLineBreak(oldText);
-                    newText = modifyData.call(this, oldText);
+                    newText = modifyData(oldText);
                 }
             } else {
                 //oldText = this.replaceLineBreak(oldText);
-                newText = modifyData.call(this, oldText);
+                newText = modifyData(oldText);
```

Comment:
Das Problem mit dem `this` argument habe ich gelöst indem die methoden nun als "lambda" deklariert sind, also 
`    artikelUndKontraktionen = (s: string): string => {`
anstatt
`    artikelUndKontraktionen(s: string): string {`
Dadurch ist das `this` gebindet.